# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/MediaEngineWebRTCAudio.cpp
# Commit: 3d0e6a5fdfb9
# Full Hash: 3d0e6a5fdfb9f2a0618e4dcd8637b2369259f09e
# Author: serge-sans-paille <sguelton@mozilla.com>
# Date: 2024-11-06 16:51:31
# Regressor Bug: 1929213
# File Overlap Count: 1
# Description:
#   Bug 1929213 - Use std::clamp instead of min/max chain in dom/media/webrtc r=bwc
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227936
# ==============================================================================

diff -r 66fa16b4e99e -r 3d0e6a5fdfb9 dom/media/webrtc/MediaEngineWebRTCAudio.cpp
--- a/dom/media/webrtc/MediaEngineWebRTCAudio.cpp	Wed Nov 06 10:14:24 2024 +0000
+++ b/dom/media/webrtc/MediaEngineWebRTCAudio.cpp	Wed Nov 06 10:15:25 2024 +0000
@@ -142,7 +142,7 @@
   // Get the number of channels asked for by content, and clamp it between the
   // pref and the maximum number of channels that the device supports.
   prefs.mChannels = c.mChannelCount.Get(std::min(prefs.mChannels, maxChannels));
-  prefs.mChannels = std::max(1, std::min(prefs.mChannels, maxChannels));
+  prefs.mChannels = std::clamp(prefs.mChannels, 1, maxChannels);
 
   LOG("Mic source %p Audio config: aec: %s, agc: %s, noise: %s, channels: %d",
       this, prefs.mAecOn ? "on" : "off", prefs.mAgcOn ? "on" : "off",